Plugin v1.0.0-rc.5 crashed

* GKE: 1.25
* Terraform Provider: v1.0.0-rc.5
* Flux: v2.0.0-rc.5

Fresh Flux install with Terraform, the Terraform provider just crashed.

```
Terraform will perform the following actions:

  # flux_bootstrap_git.flux[0] will be created
  + resource "flux_bootstrap_git" "flux" {
      + cluster_domain       = "cluster.local"
      + components           = [
          + "helm-controller",
          + "kustomize-controller",
          + "notification-controller",
          + "source-controller",
        ]
      + components_extra     = [
          + "image-automation-controller",
          + "image-reflector-controller",
        ]
      + id                   = (known after apply)
      + interval             = "1m0s"
      + log_level            = "info"
      + namespace            = "flux-system"
      + network_policy       = true
      + path                 = "gitops/clusters/kiln-gcp-devnet-europe-west1"
      + registry             = "ghcr.io/fluxcd"
      + repository_files     = (known after apply)
      + secret_name          = "flux-system"
      + version              = "v2.0.0-rc.5"
      + watch_all_namespaces = true
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

flux_bootstrap_git.flux[0]: Creating...

╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-flux_v1.0.0-rc.5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1a95d58]

goroutine 93 [running]:
github.com/fluxcd/terraform-provider-flux/internal/provider.(*providerResourceData).GetKubernetesClient(0x221f960?)
	github.com/fluxcd/terraform-provider-flux/internal/provider/provider_resource_data.go:64 +0x18
github.com/fluxcd/terraform-provider-flux/internal/provider.(*bootstrapGitResource).Create(0xc00061a188, {0x221f960, 0xc000828fc0}, {{{{0x2228460, 0xc000837740}, {0x1c4e8c0, 0xc000836750}}, {0x222beb0, 0xc000414b90}}, {{{0x2228460, ...}, ...}, ...}, ...}, ...)
	github.com/fluxcd/terraform-provider-flux/internal/provider/resource_bootstrap_git.go:350 +0x298
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).CreateResource(0xc000626160, {0x221f960, 0xc000828fc0}, 0xc0004cb3a8, 0xc0004cb348)
	github.com/hashicorp/terraform-plugin-framework@v1.2.0/internal/fwserver/server_createresource.go:97 +0x598
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ApplyResourceChange(0xc0004cb500?, {0x221f960, 0xc000828fc0}, 0xc000435360, 0xc0004cb500)
	github.com/hashicorp/terraform-plugin-framework@v1.2.0/internal/fwserver/server_applyresourcechange.go:54 +0x4a8
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ApplyResourceChange(0xc000626160, {0x221f960?, 0xc000828e70?}, 0xc0004352c0)
	github.com/hashicorp/terraform-plugin-framework@v1.2.0/internal/proto6server/server_applyresourcechange.go:52 +0x41a
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0xc000389f40, {0x221f960?, 0xc0008284b0?}, 0xc0003fc310)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov6/tf6server/server.go:816 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x1e21fa0?, 0xc000389f40}, {0x221f960, 0xc0008284b0}, 0xc0003fc2a0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000726960, {0x2229218, 0xc000602680}, 0xc0008237a0, 0xc0008a3770, 0x31832c0, 0x0)
	google.golang.org/grpc@v1.51.0/server.go:1340 +0xd33
google.golang.org/grpc.(*Server).handleStream(0xc000726960, {0x2229218, 0xc000602680}, 0xc0008237a0, 0x0)
	google.golang.org/grpc@v1.51.0/server.go:1713 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.51.0/server.go:963 +0x28a

Error: The terraform-provider-flux_v1.0.0-rc.5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

--- a/internal/provider/resource_bootstrap_git.go
+++ b/internal/provider/resource_bootstrap_git.go
@@ -82,6 +82,9 @@ const (
 	rfc1123DomainError = "a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character"
 	tolerationKeyRegex = `^[A-Za-z0-9]([A-Za-z0-9._-]*)$`
 	tolerationKeyError = "a toleration key must begin with a letter or number, and may contain letters, numbers, hyphens, dots, and underscores."
+
+	missingConfiguration                   = "Missing configuration"
+	bootstrapGitResourceMissingConfigError = "Git and Kubernetes configuration not found"
 )
 
 type bootstrapGitResourceData struct {
@@ -298,6 +301,10 @@ func (r *bootstrapGitResource) Schema(ctx context.Context, req resource.SchemaRe
 }
 
 func (r bootstrapGitResource) ModifyPlan(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) {
+	if r.prd == nil {
+		resp.Diagnostics.AddError(missingConfiguration, bootstrapGitResourceMissingConfigError)
+	}
+
 	// Skip when deleting or on initial creation
 	if req.State.Raw.IsNull() || req.Plan.Raw.IsNull() {
 		return
@@ -332,6 +339,10 @@ func (r bootstrapGitResource) ModifyPlan(ctx context.Context, req resource.Modif
 
 // TODO: If kustomization file exists and not all resource files exist bootstrap will fail. This is because kustomize build is run.
 func (r *bootstrapGitResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	if r.prd == nil {
+		resp.Diagnostics.AddError(missingConfiguration, bootstrapGitResourceMissingConfigError)
+	}
+
 	var data bootstrapGitResourceData
 	diags := req.Plan.Get(ctx, &data)
 	resp.Diagnostics.Append(diags...)
@@ -442,6 +453,10 @@ func (r *bootstrapGitResource) Create(ctx context.Context, req resource.CreateRe
 // TODO: Consider if more value reading should be done here to detect drift. Similar to how import works.
 // TODO: Resources in the cluster should be verified to exist. If not resource id should be set to nil. This is to detect changing clusters.
 func (r *bootstrapGitResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	if r.prd == nil {
+		resp.Diagnostics.AddError(missingConfiguration, bootstrapGitResourceMissingConfigError)
+	}
+
 	var data bootstrapGitResourceData
 	diags := req.State.Get(ctx, &data)
 	resp.Diagnostics.Append(diags...)
@@ -492,6 +507,10 @@ func (r *bootstrapGitResource) Read(ctx context.Context, req resource.ReadReques
 
 // TODO: Verify Flux components after updating Git
 func (r bootstrapGitResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	if r.prd == nil {
+		resp.Diagnostics.AddError(missingConfiguration, bootstrapGitResourceMissingConfigError)
+	}
+
 	var data bootstrapGitResourceData
 	diags := req.Plan.Get(ctx, &data)
 	resp.Diagnostics.Append(diags...)
@@ -580,6 +599,10 @@ func (r bootstrapGitResource) Update(ctx context.Context, req resource.UpdateReq
 }
 
 func (r bootstrapGitResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	if r.prd == nil {
+		resp.Diagnostics.AddError(missingConfiguration, bootstrapGitResourceMissingConfigError)
+	}
+
 	var data bootstrapGitResourceData
 	diags := req.State.Get(ctx, &data)
 	resp.Diagnostics.Append(diags...)
@@ -664,6 +687,10 @@ func (r bootstrapGitResource) Delete(ctx context.Context, req resource.DeleteReq
 
 // TODO: Validate Flux installation before proceeding with import
 func (r *bootstrapGitResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	if r.prd == nil {
+		resp.Diagnostics.AddError(missingConfiguration, bootstrapGitResourceMissingConfigError)
+	}
+
 	kubeClient, err := r.prd.GetKubernetesClient()
 	if err != nil {
 		resp.Diagnostics.AddError("Kubernetes Client", err.Error())

